doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -1,102 +1,120 @@
-=================================
-Term rewriting and simplification
-=================================
+=========================
+Reasoning with equalities
+=========================
 
-.. _rewritingexpressions:
+There are multiple notions of :gdef:`equality`:
 
-Rewriting expressions
----------------------
+- :gdef:`Liebniz equality` is the standard
+  way to define equality in Coq and the Calculus of Inductive Constructions, which is in
+  terms of a relation.  The standard library defines `eq` this way:
```

Comment:
Maybe you should tell me.  Is `eq` considered a mathematical relation, a proposition or what?  `eq` is a `Prop`.  I was thinking of a relation R(a, b) such that if R contains (a, b) then a and b are equal.  Probably not phrased correctly.

`eq` is generally bound to the notation `a = b`, right?

Reply:
Right. And `eq` is a binary relation and applied to its arguments it is a proposition. "contains" is probably not appropriate since it refers to the way relations are usually defined in set theory (as sets, more precisely as subsets of the cartesian product of two sets for the case of a binary relation), whereas in type theory relations are functions (whose output is a proposition).

Reply:
Anyway, this intro now looks very good to me!

Reply:
I changed to use the right terminology:

```which is in terms of a binary relation, i.e. a binary function that returns `True` if its arguments are equal and otherwise `False`.```

Reply:
Is the True/False part right?

Reply:
No. This formulation is not correct. This is a function that "returns" a proposition (in the sense of the type after the last arrow), but not a function that computes (thus it doesn't actually ever "returns" any value such as `True` / `False`---which are not even primitive Coq values BTW).

Said otherwise, while `eq` is a binary relation, `eq x y` (i.e. `eq` fully applied) is a proposition. A proposition can be proved to be true, proved to be false, impossible to prove true or false, etc.